Sources/NIO/EventLoop.swift

```diff
@@ -89,6 +89,13 @@ public protocol EventLoop: EventLoopGroup {
     ///     - task: The closure that will be submitted to the `EventLoop` for execution.
     /// - returns: `EventLoopFuture` that is notified once the task was executed.
     func submit<T>(_ task: @escaping () throws -> T) -> EventLoopFuture<T>
+    
+    /// Submit a given async task to start execution. Once the execution is complete the given async task has been started and the returned `EventLoopFuture` will be notified with the result of execution once the task is finished.
+    ///
+    /// - parameters:
+    ///     - asyncTask: The closure that will be submitted to the `EventLoop` to start execution.
+    /// - returns: `EventLoopFuture` that is notified once the task was executed.
+    func submitAsync<T>(_ asyncTask: @escaping () throws -> EventLoopFuture<T>) -> EventLoopFuture<T>
```

Comment:
@karim-elngr couldn't this be just done as any extension ? I think there is no need to add it to the protocol itself.